rusoto/core/src/signature.rs

```diff
@@ -93,10 +95,22 @@ impl SignedRequest {
         &self.headers
     }
 
+    pub fn scheme(&self) -> String {
+        match self.scheme {
+            Some(ref p) => p.to_string(),
+            None => {
+                match self.region {
+                    Region::Custom(_) => "http".to_owned(),
```

Comment:
This is fine I think; won't the `Some(...)` branch be matched if `Region::Custom(...)` is given a scheme in the URL?